components/content/Text/Text.css

```diff
@@ -0,0 +1,188 @@
+/* Font */
+
+.diamond-text-font-body {
+  font-size: var(--diamond-font-size-base);
+  font-weight: var(--diamond-font-weight-base);
+  line-height: var(--diamond-font-line-height);
+  text-wrap: pretty;
+}
+
+.diamond-text-font-heading,
+.diamond-text-font-h1,
+.diamond-text-font-h2,
+.diamond-text-font-h3,
+.diamond-text-font-h4 {
+  font-weight: var(--diamond-font-weight-bold);
+  line-height: var(--diamond-font-line-height-sm);
+  text-wrap: balance;
+}
+
+.diamond-text-font-h1 {
+  font-size: var(--diamond-font-size-h1);
+}
+
+.diamond-text-font-h2 {
+  font-size: var(--diamond-font-size-h2);
+}
+
+.diamond-text-font-h3 {
+  font-size: var(--diamond-font-size-h3);
+}
+
+.diamond-text-font-h4 {
+  font-size: var(--diamond-font-size-h4);
+}
+
+/* Size */
```

Comment:
.diamond-text-size-* should modify font-size only, falling back to the line height on whatever element they're placed on, the diamond-text-font-* can handle modifying line-height